src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -755,7 +756,13 @@ protected void exceptionCaught(Throwable cause) {
                 LOG.info("An executor rejected a read or write operation on the ClientToProxyConnection (this is normal if the proxy is shutting down). Message: " + cause.getMessage());
                 LOG.debug("A RejectedExecutionException occurred on ClientToProxyConnection", cause);
             } else {
-                LOG.error("Caught an exception on ClientToProxyConnection", cause);
+                ExceptionHandler exHandler = proxyServer.getClientToProxyExHandler();
+                if (exHandler != null) {
+                    LOG.debug("Custom exception handler is used to process the cause");
```

Comment:
please log the handler so it's easier to debug e.g. `LOG.debug("Custom exception handler '" + ex.toString() + "' invoked", cause);`